Public/Src/Cache/ContentStore/Distributed/NuCache/ContentLocationEntry.cs

```diff
@@ -101,6 +102,31 @@ public static ContentLocationEntry TryCreateFromRedisValue(byte[] data, UnixTime
             return FromRedisValue(data, lastAccessTime);
         }
 
+        /// <summary>
+        /// Serializes an instance into a binary stream.
+        /// </summary>
+        public void Serialize(BuildXLWriter writer)
+        {
+            writer.WriteCompact(ContentSize);
+            Locations.Serialize(writer);
+            writer.Write(CreationTimeUtc);
+            long lastAccessTimeOffset = LastAccessTimeUtc.Value - CreationTimeUtc.Value;

```

Comment:
What is reason for serializing in this way? Why not just write out both times?

Reply:
This is not the format I decided upon, but what has been used by the ContentLocationDatabase thus far. Since it can't be changed for backwards-compatibility, nothing can be done except for changing the format, which would require backwards compatibility and hence be more code for the same task.

A wild guess at why this is stored in this way is it may take less space to store the long than the entire timestamp.

---
In reply to: [301862358](https://github.com/microsoft/BuildXL/pull/571#discussion_r301862358) [](ancestors = 301862358)

Reply:
Julian's guess is correct. This is for every entry in the db so space savings are very important here.

---
In reply to: [302176571](https://github.com/microsoft/BuildXL/pull/571#discussion_r302176571) [](ancestors = 302176571,301862358)